### ISSUE TITLE
Please publish a proper source release tarball

### ISSUE THREAD

**Issue by rhertzog** [OCA/OCB#718]
I'm looking into packaging OCB for Debian. Unfortunately the raw content of the git repository can't be processed like any usual python project. setup.py is not working out of the box, it only works on the tarball generated by setup/package.py (which moves "addons/*" into "odoo/addons/").

Can you please generate proper release tarballs and make them available either on pypi, on github (you can upload release tarball, see [api doc on releases](https://developer.github.com/v3/repos/releases/)), or somewhere else ?

Odoo is doing it here for instance: 
https://nightly.odoo.com/11.0/nightly/src/

Thank you in advance.

**Comment by pedrobaeza** [comment]
We don't have plans for now of doing this, but if you provide a proper script that is capable to do it, we can put it on our servers for processing it after the synchronization with upstream Odoo.

**Comment by rhertzog** [comment]
You already have the script: `setup/package.py --no-testing --no-debian --no-rpm --no-windows`
Then you just need to put the generated tarball somewhere online.

**Comment by pedrobaeza** [comment]
OK, thanks for the info. I don't know anything about how to manipulate that thing, so I prefer someone else say about it.

**Comment by JordiBForgeFlow** [comment]
@sbidoul Perhaps this is something that can be done on a nightly process, similar to what we're currently doing with the publication of pip packages for OCA modules?

**Comment by sbidoul** [comment]
Yes a nightly job on the OCA server is the way to go.

@rhertzog there are probably things to adapt in the scripts though. From the top of my head that would include renaming the package to something else than odoo and adding some metadata such as "provides" odoo and "conflicts" odoo, since OCB is meant as a drop-in replacement for odoo. 

Could you help with a PR for that part?

**Comment by rhertzog** [comment]
Do you want to do this in the script or would you just rather do this directly in the OCB git repository? (i.e. simply update setup.py). I think updating the setup.py in the git repository directly would be cleaner and more logical.

**Comment by sbidoul** [comment]
> updating the setup.py in the git repository directly would be cleaner and more logical.

I agree. But we need the advice of @OCA/backport-maintainers on this.

**Comment by StefanRijnhart** [comment]
I'd like to argue against packaging OCB for Debian entirely until we have the tests under control https://travis-ci.org/OCA/OCB/branches

**Comment by JordiBForgeFlow** [comment]
@mreficent can you work to greenify OCB? This will be a priority for us

**Comment by StefanRijnhart** [comment]
Thanks @jbeficent. @mreficent can you mention me on any PRs that you file for that purpose? If we can get to green, then modifying the setup.py as necessary seems like a good option to me.

**Comment by MiquelRForgeFlow** [comment]
I think that v8 and v11 both fail because it takes too long to test everything.

    The job exceeded the maximum time limit for jobs, and has been terminated.

I think the limit is in 50 minutes. Could we put a longer limit?

**Comment by pedrobaeza** [comment]
That can't be possible for free builds.

I think it can be solved splitting the modules to test in several builds.

**Comment by StefanRijnhart** [comment]
I proposed something in this thread https://github.com/OCA/OCB/issues/639 about splitting up the build.

**Comment by pedrobaeza** [comment]
Yeah, that was my inspiration for the answer. I think it's a good solution.

**Comment by pedrobaeza** [comment]
@rhertzog I assume this is already done, isn't it?

**Comment by JordiBForgeFlow** [comment]
@StefanRijnhart seeing that after several improvements made from you, @moylop260 and @mreficent the OCB branches are beeing greenified https://travis-ci.org/OCA/OCB/branches. 

@sbidoul has offered to start the changes required to produce the nightly release tarballs, including making changes to setup.py.

**Comment by rhertzog** [comment]
@pedrobaeza What are you referring to? If your "this" refers to this ticket, then no, it's not yet done. For the Debian packaging I used the release tarball of odoo directly because OCB currently contains no functional changes (only travis integration changes AFAIK).

**Comment by JordiBForgeFlow** [comment]
@rhertzog @sbidoul is now implementing the changes to the setup.py

**Comment by sbidoul** [comment]
Nightlies should come up soon: https://github.com/OCA/maintainer-tools/pull/338

**Comment by sbidoul** [comment]
@rhertzog ocb sdists are now uploaded nightly to https://wheelhouse.odoo-community.org/ocb-nightly/

**Comment by rhertzog** [comment]
Thanks @sbidoul Feel free to close this issue then.

**Comment by MiquelRForgeFlow** [comment]
v8 in Travis is red  🔴 
```yml
+ phantomjs-prebuilt@2.1.16
added 84 packages in 4.023s
Downloading Odoo
fatal: destination path '/home/travis/OCB-8.0' already exists and is not an empty directory.
The command "travis_install_nightly" failed and exited with 128 during .
```

**Comment by moylop260** [comment]
@mreficent 
FYI should be fixed the travis red of branch 8.0 from:
  - https://github.com/OCA/OCB/pull/747

**Comment by enricostano** [comment]
Maybe I'm missing something but probably would be nice to have nighly info somewhere in the documentation of this repo. Maybe we can replace the `nightly` badge in the README or something like that.

In any case, many thanks for the OCB nightly builds! :heart:
